packages/headless/src/controllers/commerce/core/result-list/headless-core-interactive-result.test.ts

```diff
@@ -0,0 +1,58 @@
+import {configuration} from '../../../../app/common-reducers';
+import {productClick} from '../../../../features/commerce/context/product/product-actions';
+import {buildMockCommerceState} from '../../../../test/mock-commerce-state';
+import {
+  buildMockCommerceEngine,
+  MockedCommerceEngine,
+} from '../../../../test/mock-engine-v2';
+import {buildCoreInteractiveResult} from './headless-core-interactive-result';
+
+jest.mock('../../../../features/commerce/context/product/product-actions');
+
+describe('core interactive result', () => {
+  let engine: MockedCommerceEngine;
+
+  const product = {
+    productId: '1',
+    name: 'clicked on product',
+    price: 17.99,
+  };
+
+  function initializeInteractiveResult(delay?: number) {
+    buildCoreInteractiveResult(engine, {
+      options: {
+        product,
+        position: 1,
+        selectionDelay: delay,
```

Comment:
To learn a little bit more about this, why do we want to delay selection sometimes?

Reply:
With `select`, one can trigger a click immediately. The core controller offers the `delay` option to handle DOM touchstart and DOM touchend events. In other words, it helps handle longpresses:
https://github.com/coveo/ui-kit/blob/a5296ec9640f9453b6b1b661abf04621b523b8fc/packages/headless/src/controllers/core/interactive-result/headless-core-interactive-result.ts#L88-L96
More info in the docs here (I found the example snippet up top helpful!): https://docs.coveo.com/en/headless/latest/reference/search/controllers/interactive-result/#begindelayedselect